### PR TITLE
Use HTTPProxy settings from containers.conf

### PR DESCRIPTION
**Pull request by rhatdan**
This PR takes the settings from containers.conf and uses
them.  This works on the podman local but does not fix the
issue for podman remote or for APIv2.  We need a way
to specify optionalbooleans when creating containers.

Fixes: https://github.com/containers/podman/issues/8843

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: pkg/specgen/generate/container.go [hunk 1]
```diff
@@ -100,13 +99,9 @@ func CompleteSpec(ctx context.Context, r *libpod.Runtime, s *specgen.SpecGenerat
...
 		return nil, err
 	}
-	// First transform the os env into a map. We need it for the labels later in
-	// any case.
-	osEnv, err := envLib.ParseSlice(os.Environ())
-	if err != nil {
-		return nil, errors.Wrap(err, "error parsing host environment variables")
-	}
 
+	rtc.Containers.EnvHost = s.EnvHost
```
Comment (mheon) [inline pkg/specgen/generate/container.go:103]: This doesn't seem right. We should not be modifying the runtime config once Podman is running - it's asking for all sorts of wierdness in `podman system service` when multiple calls come in at the same time.

Comment (rhatdan) [reply pkg/specgen/generate/container.go:103]: This is data specific to the container, not to the general config.

Comment (rhatdan) [reply pkg/specgen/generate/container.go:103]: I think I am mistaken, opened a PR for containers/common.

**Review by vrothberg — COMMENTED** [review]
Needs tests to make sure the flag is really working as intended.

Do we really need to worry about the remote case?  If I understand it correctly, we just expose a `boolean` on the CLI which should also be treated in the remote API.  If set, the backend/server would append the proxy.  Or is it intended to be client side.

**Comment by rhatdan** [comment]
There are httpd_proxy tests already, the only change here was to fix the default in containers.conf to match the reality. But the reality of this patch is just to get the buildah code updated.  I would rather get this in, and then add tests of containers.conf later.

**Review by vrothberg — COMMENTED** [review]
LGTM

**Review by giuseppe — APPROVED** [review]
/lgtm

**Pull request merged**